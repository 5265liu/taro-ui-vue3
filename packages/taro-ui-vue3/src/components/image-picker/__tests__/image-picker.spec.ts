import { mount } from '@vue/test-utils'
import AtImagePicker from '../index.vue'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtImagePicker, {
    slots,
    props: { ...values },
  })
}

describe('AtImagePicker Snap', () => {
  const files = [
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg',
      id: '2122',
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    },
    {
      url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
      id: '2121',
    },
  ]

  it('render initial AtImagePicker', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtImagePicker -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtImagePicker -- props customStyle', () => {
    const wrapper = factory({ customStyle: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtImagePicker -- props files', () => {
    const wrapper = factory({ files: files })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtImagePicker -- props length', () => {
    const wrapper = factory({ files: files, length: 5 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtImagePicker -- props mode', () => {
    const wrapper = factory({ files: files, mode: 'top' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtImagePicker -- props showAddBtn', () => {
    const wrapper = factory({ files: files, showAddBtn: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtImagePicker -- props multiple', () => {
    const wrapper = factory({ files: files, multiple: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})
