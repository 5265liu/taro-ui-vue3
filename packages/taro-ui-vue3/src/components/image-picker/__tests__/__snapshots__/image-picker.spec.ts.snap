// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtImagePicker Snap render AtImagePicker -- props className 1`] = `
<view
  class="at-image-picker"
>
  
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      
      <!-- + add bar: choose button -->
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
        
      </view>
      
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    
  </view>
  
</view>
`;

exports[`AtImagePicker Snap render AtImagePicker -- props customStyle 1`] = `
<view
  class="at-image-picker"
  customstyle="color:red;"
>
  
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      
      <!-- + add bar: choose button -->
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
        
      </view>
      
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    
  </view>
  
</view>
`;

exports[`AtImagePicker Snap render AtImagePicker -- props files 1`] = `
<view
  class="at-image-picker"
>
  
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      
      <!-- + add bar: choose button -->
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
        
      </view>
      
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    
  </view>
  
</view>
`;

exports[`AtImagePicker Snap render AtImagePicker -- props length 1`] = `
<view
  class="at-image-picker"
>
  
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      
      <!-- + add bar: choose button -->
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
        
      </view>
      
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    
  </view>
  
</view>
`;

exports[`AtImagePicker Snap render AtImagePicker -- props mode 1`] = `
<view
  class="at-image-picker"
>
  
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="top"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      
      <!-- + add bar: choose button -->
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
        
      </view>
      
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    
  </view>
  
</view>
`;

exports[`AtImagePicker Snap render AtImagePicker -- props multiple 1`] = `
<view
  class="at-image-picker"
>
  
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      
      <!-- + add bar: choose button -->
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
        
      </view>
      
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    
  </view>
  
</view>
`;

exports[`AtImagePicker Snap render AtImagePicker -- props showAddBtn 1`] = `
<view
  class="at-image-picker"
>
  
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    
  </view>
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <view
        class="at-image-picker__item"
      >
        <view
          class="at-image-picker__remove-btn"
        />
        <image
          class="at-image-picker__preview-img"
          mode="aspectFill"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </view>
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    
  </view>
  
</view>
`;

exports[`AtImagePicker Snap render initial AtImagePicker 1`] = `
<view
  class="at-image-picker"
>
  
  <view
    class="at-image-picker__flex-box"
  >
    
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      
      <!-- + add bar: choose button -->
      <view
        class="at-image-picker__item at-image-picker__choose-btn"
      >
        
        <view
          class="add-bar"
        />
        <view
          class="add-bar"
        />
        
      </view>
      
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    <view
      class="at-image-picker__flex-item"
    >
      <!-- image thumb -->
      <!--v-if-->
    </view>
    
  </view>
  
</view>
`;
